### PR TITLE
Overhaul package, add optional twisted based client. Replaces #76

### PR DESCRIPTION
**Pull request by rnixx**
This pull request replaces #76.

* Create mpd package, move code from mpd.py to mpd/base.py
* Import public API in mpd/__init__.py
* Move more command result parsers to MPDClientBase
* Commands are now registered with mpd_commands decorator on MPDClientBase
* Add some more parser tests
* Add mpd/twisted.py module. Contains an MPDProtocol implementation. The installation of twisted is NOT mandatory, just results in an ImportError if you try to use it and twisted is not installed.
* Run tests with coverage

### CODE REVIEW

**Comment by Mic92** [comment]
Merged in https://github.com/Mic92/python-mpd2/commits/twisted with some additional changes.
One thing missing, is at least one test to get some basic coverage (for instance detect breakage with future version of twisted).

File: mpd/twisted.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            deferred.addCallback(parser)
+            # command list, attach handler for collecting command list results
+            if self._command_list:
+                deferred.addCallback(self._parse_command_list_item)
+        return deferred
+
+    def _write_command(self, command, args=[]):
+        parts = [command]
+        parts += ['"{}"'.format(escape(arg.encode('utf-8')) \
+            if isinstance(arg, unicode) else str(arg)) for arg in args]
```
Comment (Mic92) [inline mpd/twisted.py]: so is this code tested with python2 only?

Comment (rnixx) [inline mpd/twisted.py]: Yes, I need to fire up a python 3 instance and fix the unicode issue. Thanks for the hint.

**Comment by rnixx** [comment]
I'll add some tests.

File: mpd/twisted.py (hunk 1 above)
Comment (Mic92) [inline mpd/twisted.py]: you can use: `encode_str` from base.py

**Comment by Mic92** [comment]
This does have to cover 100%, but it should at least include basic command encoding.

**Comment by Mic92** [comment]
merged in 5bf0e0643d80a5c1431d19fdcb05b25a0f930e24

**Pull request closed without merging**